src/Safebox.sol

```diff
@@ -0,0 +1,197 @@
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity ^0.8.16;
+
+import {VatAbstract} from "dss-interfaces/dss/VatAbstract.sol";
+import {GemAbstract} from "dss-interfaces/ERC/GemAbstract.sol";
+import {SafeboxLike} from "./SafeboxLike.sol";
+
+/**
+ * @author amusingaxl
+ * @title A safebox for ERC-20 tokens.
+ * @notice
+ * - The `owner`(MakerDAO Governance) is in full control of how much and when it can send tokens to `recipient`.
+ *     - MakerDAO Governance could add other owners if required in the future (i.e.: automation of the Safebox balance).
+ * - If MakerDAO governance ever executes an Emergency Shutdown, anyone can send tokens to `recipient`.
+ *     - This prevents tokens being stuck in this contract when the governance smart contract is no longer operational.
+ * - The `custodian` cooperation is required whenever the `owner` wants to update the `recipient`.
+ */
+contract Safebox is SafeboxLike {
+    /// @notice MCD Vat module.
+    VatAbstract public immutable override vat;
+    /// @notice The ERC-20 token to be held in this contract.
+    GemAbstract public immutable override token;
+
+    /// @notice Addresses with owner access on this contract. `wards[usr]`
+    mapping(address => uint256) public override wards;
+    /// @notice Addresses with custodian access on this contract. `custodians[usr]`
+    mapping(address => uint256) public override custodians;
+
+    /// @notice The recipient for the tokens held in this contract.
+    address public override recipient;
+    /// @notice Reference to the new recipient when it is to be changed.
+    address public override pendingRecipient;
+
+    modifier auth() {
+        require(wards[msg.sender] == 1, "Safebox/not-ward");
+        _;
+    }
+
+    modifier onlyCustodian() {
+        require(custodians[msg.sender] == 1, "Safebox/not-custodian");
+        _;
+    }
+
+    /**
+     * @param _vat The MCD vat module.
+     * @param _token The ERC-20 token to be held in this contract.
+     * @param _owner The safebox owner.
+     * @param _custodian The safebox custodian.
+     * @param _recipient The recipient for tokens in the safebox.
+     */
+    constructor(address _vat, address _token, address _owner, address _custodian, address _recipient) {
+        require(_recipient != address(0), "Safebox/invalid-recipient");
+
+        wards[_owner] = 1;
```

Comment:
Note that common behavior for MakerDao contracts is to only rely msg.sender in the constructor.

Reply:
Yeah, we are aware of that, however, since Coinbase will be handling the deployment, we want to avoid them having to `deny` themselves afterwards.

Reply:
Just to clarify, we are gonna provide them the parameters for deployment and then verify the contract after it's deployed. If there is a mismatch, the verification will fail and we will know.

Also the variable are public, so they could be easily checked after deployment & verification.